server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java

```diff
@@ -203,12 +196,9 @@ public void cancel(TaskId taskId) {
     public void close() {
         Optional.ofNullable(receiverHandle).ifPresent(Disposable::dispose);
         Optional.ofNullable(receiver).ifPresent(RabbitMQExclusiveConsumer::close);
-        Optional.ofNullable(sender).ifPresent(Sender::close);
         Optional.ofNullable(sendCancelRequestsQueueHandle).ifPresent(Disposable::dispose);
         Optional.ofNullable(cancelRequestListenerHandle).ifPresent(Disposable::dispose);
-        Optional.ofNullable(sender).ifPresent(Sender::close);
         Optional.ofNullable(cancelRequestSender).ifPresent(Sender::close);
         Optional.ofNullable(cancelRequestListener).ifPresent(Receiver::close);
-        channelPool.close();
```

Comment:
why don't you need anymore to close these resources?

Reply:
Will be closed by guice

Reply:
how is it called it tests?

Reply:
The channelPool is no longer initialized within this class but directly in the testing @BeforeEach code hence it's no longer the responsability of this code to close it but of the testing code in a @AfterEach.